darktable-org/dtorg
# migrate website to hugo

Initial port of the darktable website to [hugo](https://gohugo.io/). See https://github.com/darktable-org/dtorg/issues/113 for background.

I've been able to convert most of the markdown and layout files to a hugo compatible format. There are numerous ways to accomplish things with hugo, so I'd love recommendations if there's a better/easier way to do something here.

Notes:
- Uses hugo version `0.83.1`
- Uses GitHub actions to deploy a Netlify preview at https://hugo-robbymilo--darktable.netlify.app (can consider using [Netlify's open source plan](https://www.netlify.com/legal/open-source-policy/))
- Adds responsive images for some of the ledes/backgrounds. Note I am not committing most of the rendered thumbnails to git despite hugo's recommendation.
- I've kept the old theme files and .py scripts for reference, these can probably be removed.

To-do list for parity with the current website:

- [x] Handle posts with different slugs - either redirect old urls (alias?) or add a `slug` param with the proper value. For example, the current site renders as [/2020/08/darktable-321-released/](https://www.darktable.org/2020/08/darktable-321-released/), but hugo renders as [/2020/08/darktable-3.2.1-released/](https://hugo-robbymilo--darktable.netlify.app/2020/08/darktable-3.2.1-released/)
- [ ] Current search allows for searching the lua api - where are these located? Are these pages a different site?
- [x] Correct footer menu links
- [ ] Check for correct rendering of markdown - I've spot checked some of the articles and it looks like there may be some differences in how hugo (goldmark) renders markdown. For example, hugo does not render markdown within HTML. I've solved this by adding a shortcode. See https://github.com/robbymilo/dtorg/blob/robbymilo/hugo/content/blog/2020-08-10-darktable-3.2/index.md for an example with shortcodes.
- [ ] Check for broken links/images

To test locally:
- Make sure you have the correct hugo version installed. Run `make server` and navigate to localhost:1313

Feel free to add to the to-do list, or even better a to-do and solution :) 

cc @paperdigits
